src/GUI/GUI.cpp

```diff
@@ -144,6 +144,16 @@ namespace RC::GUI
                     BPMods::render();
                     ImGui::EndTabItem();
                 }
+
+                std::lock_guard<std::mutex> guard(m_tabs_mutex);
+                for (const auto& tab : m_tabs)
+                {
+                    if (ImGui::BeginTabItem(to_string(tab->TabName).c_str()))
+                    {
+                        tab->render();
+                    }
+                }
```

Comment:
You should wrap the mutex & for-loop in its own scope since we're only looking to protect `m_tabs` and not the entire `update` function:
```suggestion
                {
                    std::lock_guard<std::mutex> guard(m_tabs_mutex);
                    for (const auto& tab : m_tabs)
                    {
                        if (ImGui::BeginTabItem(to_string(tab->TabName).c_str()))
                        {
                            tab->render();
                        }
                    }
                }
```